Gems/PhysX/Code/Source/EditorMeshColliderComponent.cpp

```diff
@@ -0,0 +1,1101 @@
+
+            AZ_Assert(shapeConfiguration && colliderConfiguration, "DisplayMeshCollider: Invalid shape-collider configuration pair");
+
+            switch (shapeConfiguration->GetShapeType())
+            {
+            case Physics::ShapeType::CookedMesh:
+            {
+                const Physics::CookedMeshShapeConfiguration* cookedMeshShapeConfiguration =
+                    static_cast<const Physics::CookedMeshShapeConfiguration*>(shapeConfiguration);
+
+                const AZ::Vector3 overallScale = Utils::GetTransformScale(GetEntityId()) * m_cachedNonUniformScale * assetScale;
+                Physics::ColliderConfiguration nonUniformScaledColliderConfiguration = *colliderConfiguration;
+                nonUniformScaledColliderConfiguration.m_position *= m_cachedNonUniformScale;
+
+                m_colliderDebugDraw.DrawMesh(debugDisplay, nonUniformScaledColliderConfiguration, *cookedMeshShapeConfiguration,
+                    overallScale, static_cast<AZ::u32>(shapeIndex));
+                break;
+            }
+            case Physics::ShapeType::Sphere:
+            {
+                const Physics::SphereShapeConfiguration* sphereShapeConfiguration =
+                    static_cast<const Physics::SphereShapeConfiguration*>(shapeConfiguration);
+
+                m_colliderDebugDraw.DrawSphere(debugDisplay, *colliderConfiguration, *sphereShapeConfiguration, assetScale);
+                break;
+            }
+            case Physics::ShapeType::Box:
+            {
+                const Physics::BoxShapeConfiguration* boxShapeConfiguration =
+                    static_cast<const Physics::BoxShapeConfiguration*>(shapeConfiguration);
+
+                m_colliderDebugDraw.DrawBox(debugDisplay, *colliderConfiguration, *boxShapeConfiguration, assetScale);
+                break;
+            }
+            case Physics::ShapeType::Capsule:
+            {
+                const Physics::CapsuleShapeConfiguration* capsuleShapeConfiguration =
+                    static_cast<const Physics::CapsuleShapeConfiguration*>(shapeConfiguration);
+
+                m_colliderDebugDraw.DrawCapsule(debugDisplay, *colliderConfiguration, *capsuleShapeConfiguration, assetScale);
+                break;
+            }
+            default:
+            {
+                AZ_Error("EditorMeshColliderComponent", false, "DisplayMeshCollider: Unsupported ShapeType %d. Entity %s, ID: %llu",
+                    static_cast<AZ::u32>(shapeConfiguration->GetShapeType()), GetEntity()->GetName().c_str(), GetEntityId());
+                break;
+            }
+            }
+        }
+    }
+
+    void EditorMeshColliderComponent::Display(
+        [[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo,
+        AzFramework::DebugDisplayRequests& debugDisplay) const
+    {
+        if (!m_colliderDebugDraw.HasCachedGeometry())
+        {
+            BuildDebugDrawMesh();
+        }
+
+        if (m_colliderDebugDraw.HasCachedGeometry())
```

Comment:
question: This check seems redundant unless `BuildDebugDrawMesh` could fail? Can that happen do we know?

Reply:
You might be right. These editor classes have evolved having multiple checks along the way, probably because they where needed at some point. So I'd rather keep it to be safe and have the same code as before.